### PR TITLE
Add ghjvprod implementation and tests

### PR DESCRIPTION
**Pull request by tmigot**
- add API doc
- doc in NLPModels.jl
- autodiff implementations
- feasibility/slack/lls implementations
- implement in test set
- consistency test

### CODE REVIEW

File: src/NLPModels.jl [hunk 1]
```diff
@@ -427,12 +427,20 @@ end
 
 function jth_hprod! end
 
+"""
+   gHv = ghjvprod(nlp, x, g, v)
+Return the vector whose i-th component is g' ∇²cᵢ(x) v.
```
Comment (dpo) [inline src/NLPModels.jl:432]: Don’t we need a blank line before the `Return` so the markdown is formatted correctly?

File: src/autodiff_model.jl [hunk 2]
```diff
@@ -241,3 +241,12 @@ function hprod!(nlp :: ADNLPModel, x :: AbstractVector, y :: AbstractVector, v :
   Hv .= ForwardDiff.derivative(t -> ForwardDiff.gradient(ℓ, x + t * v), 0)
   return Hv
 end
+
+function ghjvprod!(nlp :: ADNLPModel, x :: AbstractVector, g :: AbstractVector, v :: AbstractVector, gHv :: AbstractVector) 
+  @lencheck nlp.meta.nvar x g v
+  @lencheck nlp.meta.ncon gHv
+  increment!(nlp, :neval_hprod)
+  ℓ(x) = nlp.c(x)
```
Comment (dpo) [inline src/autodiff_model.jl:249]: Necessary?

**Review by dpo — APPROVED** [review]
Thanks!

File: src/NLPModels.jl (hunk 1 above)
Comment (tmigot) [reply src/NLPModels.jl]: You are right, thanks !

File: src/autodiff_model.jl (hunk 2 above)
Comment (tmigot) [reply src/autodiff_model.jl]: You are right, thanks !

**Pull request merged**